qos-service-impl/src/main/java/com/palantir/atlasdb/qos/config/CassandraHealthMetricMeasurement.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.qos.config;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class CassandraHealthMetricMeasurement {
+    abstract double lowerLimit();
+    abstract double upperLimit();
+    abstract Object currentValue();
+
+    boolean isMeasurementWithinLimits() {
+        double measurement;
+        try {
+            measurement = (double) currentValue();
+        } catch (ClassCastException ex) {
+            // TODO(hsaraogi): The metric is not a number, should we throw here?
+            return true;
+        }
+        return Double.compare(lowerLimit(), measurement) <= 0
```

Comment:
this might be more readable as `lowerLimit() < measurement && measurement < upperLimit()`